tools/src/block-gen/Context.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE TemplateHaskell      #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+-- | Initial context satisfing MonadBlockGen.
+
+module Context
+       ( bracketTBlockGenMode
+       ) where
+
+import           Universum
+
+import           Control.Lens         (makeLensesWith)
+import qualified Control.Monad.Reader as Mtl
+import           Ether.Internal       (HasLens (..))
+import           Mockable             (Production, currentTime)
+
+import           Pos.Block.Core       (Block, BlockHeader)
+import           Pos.Block.Slog       (SlogContext, mkSlogContext)
+import           Pos.Block.Types      (Undo)
+import           Pos.Context          (GenesisUtxo (..))
+import           Pos.Core             (Timestamp (..))
+import           Pos.DB               (MonadBlockDBGeneric (..),
+                                       MonadBlockDBGenericWrite (..), MonadDB (..),
+                                       MonadDBRead (..))
+import qualified Pos.DB               as DB
+import qualified Pos.DB.Block         as DB
+import           Pos.DB.DB            (initNodeDBs)
+import           Pos.DB.Pure          (DBPureVar, newDBPureVar)
+import           Pos.GState           (GStateContext (..), GStateContextPure)
+import qualified Pos.GState           as GS
+import           Pos.Launcher         (newInitFuture)
+import           Pos.Lrc.Context      (LrcContext (..), mkLrcSyncData)
+import           Pos.Slotting         (HasSlottingVar (..))
+import           Pos.Ssc.GodTossing   (SscGodTossing)
+import           Pos.Util.Util        (postfixLFields)
+
+data TBlockGenContext = TBlockGenContext
+    { tbgcGState      :: GStateContextPure
```

Comment:
it's the temporary solution, I described at the slack. we can implement it via PureDb and cloning